tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -52,15 +53,32 @@ class SerializableModelLibraryGenerator {
       (libraryBuilder) {
         if (classDefinition.isSealedTopNode) {
           for (var child in classDefinition.descendantClasses) {
-            libraryBuilder.directives
-                .add(Directive.part('${child.fileName}.dart'));
+            var childPath = p.relative(
+              p.joinAll([
+                ...child.subDirParts,
+                '${child.fileName}.dart',
+              ]),
```

Comment:
Suggestion: Make this an extension on `classDefinitions`.

We seem to be building this same path over and over again. Might as well make it an extension and use it everywhere in the generator.